src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -479,14 +522,157 @@ void shouldDisplaySeriesPosition_withSeriesPositionPopulated_whenBookHasSeriesPo
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        Book savedBook = bookService.findAll().get(0);
+        populateBookFormWithExistingBook(READ, savedBook);
+
+        // when
+        bookForm.bookTitle.setValue("IT");
+        bookForm.authorFirstName.setValue("Stephen");
+        bookForm.authorLastName.setValue("King");
+        bookForm.bookGenre.setValue(Genre.HORROR);
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book updatedBook = bookService.findAll().get(0);
+        Assertions.assertEquals("IT", updatedBook.getTitle());
+        Assertions.assertEquals(READ, updatedBook.getPredefinedShelf().getPredefinedShelfName());
+        Assertions.assertEquals("Stephen", updatedBook.getAuthor().getFirstName());
+        Assertions.assertEquals("King", updatedBook.getAuthor().getLastName());
+        Assertions.assertEquals(Genre.HORROR, updatedBook.getGenre());
+        Assertions.assertEquals(dateStarted, updatedBook.getDateStartedReading());
+        Assertions.assertEquals(dateFinished, updatedBook.getDateFinishedReading());
+    }
+
+
+    private static Stream<Arguments> shelfCombinations() {
+        return Stream.of(
+                Arguments.of(TO_READ, READING),
+                Arguments.of(TO_READ, READ),
+                Arguments.of(TO_READ, DID_NOT_FINISH),
+                Arguments.of(READING, DID_NOT_FINISH),
+                Arguments.of(READING, READ)
+                // TODO: these cases are not passing atm because of issue #271
+                // Arguments.of(READING, TO_READ),
+                // Arguments.of(READ, TO_READ),
+                // Arguments.of(READ, DID_NOT_FINISH),
+                // Arguments.of(READ, READING),
+                // Arguments.of(DID_NOT_FINISH, TO_READ),
+                // Arguments.of(DID_NOT_FINISH, READING),
+                // Arguments.of(DID_NOT_FINISH, READ)
+                );
+    }
+
+    /**
+     * Tests whether a book is updated in the database when the save-event is called
+     */
+    @ParameterizedTest
+    @MethodSource("shelfCombinations")
+    void shouldUpdateValuesWhenBookIsMovedBetweenShelves(ShelfName initialShelf, ShelfName newShelf) {
+        // given
+        bookForm = createBookForm(initialShelf, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+        populateBookFormWithExistingBook(newShelf, bookService.findAll().get(0));
+
+        // when
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book bookInDatabase = bookService.findAll().get(0);
+        assertBookInSpecificShelf(newShelf, bookInDatabase);
+    }
+
+    private void assertBookInSpecificShelf(ShelfName shelfName, Book bookInDatabase) {
```

Comment:
Could you please rename this method to a more descriptive name? At first glance, it isn't clear what it is testing (e.g. is it testing that a book is successfully stored in a particular shelf?).

Reply:
done